h/zssLogging.h

```diff
@@ -13,13 +13,570 @@
+#define ZSS_LOG_DISC_FIRST_NKNOWN_MSG        ZSS_LOG_DISC_FIRST_NKNOWN_MSG_ID" "ZSS_LOG_DISC_FIRST_NKNOWN_MSG_TEXT
+
+/* MVD Server (ServerStatusService) */
+
+#ifndef ZSS_LOG_FAIL_RMF_FETCH_MSG_ID
+#define ZSS_LOG_FAIL_RMF_FETCH_MSG_ID        LOG_MSG_PRFX"1050W"
+#endif
+#define ZSS_LOG_FAIL_RMF_FETCH_MSG_TEXT      "Failed to fetch RMF data, RC = %d\n"
+#define ZSS_LOG_FAIL_RMF_FETCH_MSG           ZSS_LOG_FAIL_RMF_FETCH_MSG_ID" "ZSS_LOG_FAIL_RMF_FETCH_MSG_TEXT
+
+/* Unixfile */
+
+#ifndef ZSS_LOG_UNABLE_CLOSE_MSG_ID
+#define ZSS_LOG_UNABLE_CLOSE_MSG_ID          LOG_MSG_PRFX"1200W"
+#endif
+#define ZSS_LOG_UNABLE_CLOSE_MSG_TEXT        "Could not close file. Ret: %d, Res: %d\n"
+#define ZSS_LOG_UNABLE_CLOSE_MSG             ZSS_LOG_UNABLE_CLOSE_MSG_ID" "ZSS_LOG_UNABLE_CLOSE_MSG_TEXT
+
+#ifndef ZSS_LOG_UNABLE_CREATE_MSG_ID
+#define ZSS_LOG_UNABLE_CREATE_MSG_ID         LOG_MSG_PRFX"1201W"
+#endif
+#define ZSS_LOG_UNABLE_CREATE_MSG_TEXT       "Could not create file. Ret: %d, Res: %d\n"
+#define ZSS_LOG_UNABLE_CREATE_MSG            ZSS_LOG_UNABLE_CREATE_MSG_ID" "ZSS_LOG_UNABLE_CREATE_MSG_TEXT
+
+#ifndef ZSS_LOG_UNABLE_METADATA_MSG_ID
+#define ZSS_LOG_UNABLE_METADATA_MSG_ID       LOG_MSG_PRFX"1202W"
+#endif
+#define ZSS_LOG_UNABLE_METADATA_MSG_TEXT     "Could not get metadata for file. Ret: %d, Res: %d\n"
+#define ZSS_LOG_UNABLE_METADATA_MSG          ZSS_LOG_UNABLE_METADATA_MSG_ID" "ZSS_LOG_UNABLE_METADATA_MSG_TEXT
+
+#ifndef ZSS_LOG_UNABLE_OPEN_MSG_ID
+#define ZSS_LOG_UNABLE_OPEN_MSG_ID           LOG_MSG_PRFX"1203W"
+#endif
+#define ZSS_LOG_UNABLE_OPEN_MSG_TEXT         "Could not open file. Ret: %d, Res: %d\n"
+#define ZSS_LOG_UNABLE_OPEN_MSG              ZSS_LOG_UNABLE_OPEN_MSG_ID" "ZSS_LOG_UNABLE_OPEN_MSG_TEXT
+
+#ifndef ZSS_LOG_TTYPE_NOT_SET_MSG_ID
+#define ZSS_LOG_TTYPE_NOT_SET_MSG_ID         LOG_MSG_PRFX"1203W"
+#endif
+#define ZSS_LOG_TTYPE_NOT_SET_MSG_TEXT       "Transfer type hasn't been set."
+#define ZSS_LOG_TTYPE_NOT_SET_MSG            ZSS_LOG_TTYPE_NOT_SET_MSG_ID" "ZSS_LOG_TTYPE_NOT_SET_MSG_TEXT
+
+/* Security */
+
+#ifndef ZSS_LOG_NON_STRD_GET_MSG_ID
+#define ZSS_LOG_NON_STRD_GET_MSG_ID          LOG_MSG_PRFX"1400W"
+#endif
+#define ZSS_LOG_NON_STRD_GET_MSG_TEXT        "non standard class provided for profiles GET, leaving...\n"
+#define ZSS_LOG_NON_STRD_GET_MSG             ZSS_LOG_NON_STRD_GET_MSG_ID" "ZSS_LOG_NON_STRD_GET_MSG_TEXT
+
+#ifndef ZSS_LOG_PROF_REQ_GET_MSG_ID
+#define ZSS_LOG_PROF_REQ_GET_MSG_ID          LOG_MSG_PRFX"1401W"
+#endif
+#define ZSS_LOG_PROF_REQ_GET_MSG_TEXT        "profile not provided for profiles GET, leaving...\n"
+#define ZSS_LOG_PROF_REQ_GET_MSG             ZSS_LOG_PROF_REQ_GET_MSG_ID" "ZSS_LOG_PROF_REQ_GET_MSG_TEXT
+
+#ifndef ZSS_LOG_NON_STRD_POST_MSG_ID
+#define ZSS_LOG_NON_STRD_POST_MSG_ID         LOG_MSG_PRFX"1402W"
+#endif
+#define ZSS_LOG_NON_STRD_POST_MSG_TEXT       "non standard class provided for profiles POST, leaving...\n"
```

Comment:
It's up to you. I don't think there are plans, seeing as content-wise there wouldn't be more to support, just a greater quantity of messages. But I could consolidate if you think it's worth it
I just don't see the overall positive impact.

It might even be bad. Imagine "This thing A failed to load" and "This thing B failed to load" have the same message ID... that might get confusing

Let's imagine to a tech support person, the meaning of the message now depends on what the literal text says rather than the ID. Which is what we're trying to avoid, I believe.
Thoughts? @1000TurquoisePogs

Reply:
What are the system action and operator response to these messages? If they are the same, i.e. "Request rejected" and "Correct [parm_name] and contact support if the problem hasn't been resolved" then I think it's worth having a single template.

I didn't mean arbitrary things which fall into the same template, I meant related things. For example, https://github.com/zowe/zowe-common-c/blob/master/h/crossmemory.h#L476